### PR TITLE
Feature/authorization search api

### PR DESCRIPTION
**Pull request by lumig242**
Add authorization for Metadata Search APIs.

### CODE REVIEW

**Comment by lumig242** [comment]
Build: http://builds.cask.co/browse/CDAP-DUT4540
JIRA: https://issues.cask.co/browse/CDAP-6655

File: cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java [hunk 1]
```diff
@@ -543,6 +549,38 @@ public Boolean call() throws Exception {
...
+                                                                          "value*",
+                                                                          ImmutableSet.<MetadataSearchTargetType>of());
+    // Result is empty now since Alice don't have the privileges on it
+    Assert.assertTrue(result.isEmpty());
+    // Grant access and search again
+    grantAndAssertSuccess(streamId.toEntityId(), ALICE, ImmutableSet.of(Action.READ));
+    result = metadataAdmin.searchMetadata(AUTH_NAMESPACE.getNamespace(),
+                                          "value*",
+                                          ImmutableSet.<MetadataSearchTargetType>of());
+    Assert.assertFalse(result.isEmpty());
```
Comment (bdmogal) [inline cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java]: should assert the actual expected result, not just that its non-empty

File: cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java [hunk 2]
```diff
@@ -543,6 +549,38 @@ public Boolean call() throws Exception {
...
+
+    MetadataSearchResultRecord record = (MetadataSearchResultRecord) result.toArray()[0];
```
Comment (bdmogal) [inline cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java]: ummm, why do you need casting? Can we just do result.iterator().next()?

File: cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java [hunk 3]
```diff
@@ -543,6 +549,38 @@ public Boolean call() throws Exception {
...
+    Assert.assertEquals(record.getEntityId(), streamId);
+    Metadata systemMetadata = record.getMetadata().get(MetadataScope.SYSTEM);
```
Comment (bdmogal) [inline cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java]: if you just create an expected MetadataSearchResultRecord, and assertEquals the entire record on line 575, you don't need any of these checks.

File: cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java [hunk 4]
```diff
@@ -543,6 +549,38 @@ public Boolean call() throws Exception {
     assertNoAccess(appId);
   }
 
+  @Test
+  public void testSearchMetadata() throws Exception {
+    createAuthNamespace();
+    deployApplication(AUTH_NAMESPACE.toId(), AllProgramsApp.class);
```
Comment (bdmogal) [inline cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java]: when the app is deployed, some metadata is automatically annotated to every entity in the app. might want to test some more entities than just the stream

**Comment by bdmogal** [comment]
Code looks good. Some comments about the test. After those changes, as long as build is successful and this is tested on a cluster, LGTM.

File: cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java (hunk 4 above)
Comment (lumig242) [inline cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java]: Updated.

File: cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java [hunk 5]
```diff
@@ -936,6 +941,13 @@ protected static SecureStoreManager getSecureStoreManager() {
   }
 
   /**
+   * Returns a {@link SecureStoreManager} to interact with secure storage.
```
Comment (bdmogal) [inline cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java]: MetadataAdmin

**Comment by lumig242** [comment]
Remove the change of `TestBase` class and move the test case into `appFabric`. A new build here: http://builds.cask.co/browse/CDAP-DUT4540
And tested on a cluster. Will merge.

**Pull request merged**